components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/api/model/ActionDTO.java

```diff
@@ -143,4 +168,94 @@ public ActionDTO build() {
             return new ActionDTO(this);
         }
     }
+
+    /**
+     * Builder for ActionDTO model used in Action Converter.
+     */
+    public static class BuilderForService {
+
+        private final String id;
+        private final Action.ActionTypes type;
+        private final String name;
+        private final String description;
+        private final Action.Status status;
+        private final EndpointConfig endpoint;
+        private final ActionRule rule;
+        private Map<String, Object> properties;
+
+        public BuilderForService(ActionDTO actionDTO) {
+
+            this.id = actionDTO.getId();
+            this.type = actionDTO.getType();
+            this.name = actionDTO.getName();
+            this.description = actionDTO.getDescription();
+            this.status = actionDTO.getStatus();
+            this.endpoint = actionDTO.getEndpoint();
+            this.rule = actionDTO.getActionRule();
+            this.properties = actionDTO.getProperties();
+        }
+
+        public BuilderForService(Action action) {
+
+            this.id = action.getId();
+            this.type = action.getType();
+            this.name = action.getName();
+            this.description = action.getDescription();
+            this.status = action.getStatus();
+            this.endpoint = action.getEndpoint();
+            this.rule = action.getActionRule();
+        }
+
+        public BuilderForService properties(Map<String, ActionPropertyForService> properties) {
+
+            this.properties = properties.entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().getValue()
```

Comment:
This is wrong again. We should not extract the value from the ActionProperty.
We just need to set the value of the entry which is an ActionProperty object